modules/core/src/lib/attribute.js

```diff
@@ -36,6 +36,7 @@ export default class Attribute extends BaseAttribute {
     let {defaultValue = [0, 0, 0, 0]} = opts;
     defaultValue = Array.isArray(defaultValue) ? defaultValue : [defaultValue];
 
+    this.defaultType = this.type || GL.FLOAT;
```

Comment:
Nit: Feels like this should go into `BaseAttribute` - getting hard to scan here and see that we have both `type` and `defaultType` and grok the difference.